alert/src/main/java/org/opennms/horizon/alertservice/service/AlertEventProcessor.java

```diff
@@ -105,19 +103,24 @@ public Optional<Alert> process(Event e) {
         return Optional.of(alertMapper.toProto(dbAlert));
     }
 
-    private AlertData getAlertData(Event event, AlertDefinition alertDefinition) {
+    private @Nullable AlertData getAlertData(Event event, AlertDefinition alertDefinition) {
         var reductionKey = String.format(alertDefinition.getReductionKey(), event.getTenantId(), event.getUei(), event.getNodeId());
         String clearKey = null;
         if (!Strings.isNullOrEmpty(alertDefinition.getClearKey())) {
             clearKey = String.format(alertDefinition.getClearKey(), event.getTenantId(), event.getNodeId());
         }
         TriggerEvent triggerEvent = triggerEventRepository.getReferenceById(alertDefinition.getTriggerEventId());
-        return new AlertData(reductionKey, clearKey, alertDefinition.getType(), triggerEvent.getSeverity(), triggerEvent.getCount(), triggerEvent.getOvertime(), triggerEvent.getOvertimeUnit());
+        // TODO: An alert could match multiple monitoring policies, each having their own notifications.
+        Optional<MonitorPolicy> policy = monitorPolicyRepository.findMonitoringPolicyByTriggerEvent(triggerEvent.getId());
+        if (policy.isPresent()) {
+            return new AlertData(reductionKey, clearKey, alertDefinition.getType(), triggerEvent.getSeverity(), triggerEvent.getCount(), triggerEvent.getOvertime(), triggerEvent.getOvertimeUnit(), List.of(policy.get().getId()));
+        }
+        return new AlertData(reductionKey, clearKey, alertDefinition.getType(), triggerEvent.getSeverity(), triggerEvent.getCount(), triggerEvent.getOvertime(), triggerEvent.getOvertimeUnit(), Collections.emptyList());
```

Comment:
Nit: I find these really long lines affect the readability of the code, and prefer they be wrapped. 
```suggestion
            return new AlertData(
                reductionKey,
                clearKey,
                alertDefinition.getType(),
                triggerEvent.getSeverity(),
                triggerEvent.getCount(),
                triggerEvent.getOvertime(),
                triggerEvent.getOvertimeUnit(),
                List.of(policy.get().getId()));
        }
        // Alternative wrapping
        return new AlertData(
            reductionKey, clearKey, alertDefinition.getType(), triggerEvent.getSeverity(),
            triggerEvent.getCount(), triggerEvent.getOvertime(), triggerEvent.getOvertimeUnit(),
            Collections.emptyList()
        );
```

Reply:
Given the large number of parameters here, I'd prefer to only have one return statement here, as if there are modifications to the parameters it would be easy to make a mistake.

So code like:
if (policy.isPresent()) {
policies = List.of(policy.get().getId());
} else {
policies = Collections.emptyList()
}

return new AlertData(..., policies);